packages/element/src/deprecated.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { flowRight } from 'lodash';
+import { Component } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import deprecated from '@wordpress/deprecated';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+
+/**
+ * Internal dependencies
+ */
+import createHigherOrderComponent from './create-higher-order-component';
+
+export const compose = ( ...args ) => {
+	deprecated( 'wp.element.compose', {
+		version: '3.5',
+		alternative: 'wp.compose.compose',
+	} );
+
+	return flowRight( ...args );
+};
+
+export const pure = createHigherOrderComponent( ( Wrapped ) => {
```

Comment:
Yeah, probably makes the most sense to keep those two items duplicated until they get removed 👍